server_admin/topics/admin-cli.adoc

```diff
@@ -1,1320 +1,1702 @@
 
-    $ kcadm.sh config credentials --server http://localhost:8080/auth --realm master --user admin --password admin
+The second approach only authenticates each command invocation for the duration of that invocation. This approach increases the load on the server and the time spent with roundtrips obtaining tokens. The benefit of this approach is not needing to save any tokens between invocations, which means nothing is saved to disk. This mode is used when the [command]`--no-config` argument is specified.
 
-This approach maintains an authenticated session between `kcadm` command invocations by saving the obtained access token, and
-associated refresh token, possibly other secrets as well in a private configuration file. See <<_working_with_alternative_configurations, next chapter>> for more info on configuration file.
+For example, when performing an command, we specify all the information required for authentication.
+[options="nowrap"]
+----
+$ kcadm.sh get realms --no-config --server http://localhost:8080/auth --realm master --user admin --password admin
+----
 
-Another approach is to authenticate with each command invocation for the duration of that invocation only. This approach results
-in more load on the server, and more time spent with round-trips obtaining tokens, but has a benefit of not needing to save any
-tokens between invocations, thus nothing is saved to disk. This mode is used when `--no-config` argument is specified.
-
-For example, when performing an operation we specify all the information required for authentication:
-
-    $ kcadm.sh get realms --no-config --server http://localhost:8080/auth --realm master --user admin --password admin
-
-
-See built-in help for more information on using `Admin CLI`.
-
-
-For example:
-
-    $ kcadm.sh help
-
-
-See `kcadm.sh config credentials --help` for more information about starting an authenticated session.
+Run the [command]`kcadm.sh help` command for more information on using the Admin CLI.
 
+Run the [command]`kcadm.sh config credentials --help` command for more information about starting an authenticated session.
 
 
 [[_working_with_alternative_configurations]]
 === Working with alternative configurations
 
-By default, `Admin CLI` automatically maintains a configuration file called `kcadm.config` located under user's home
-directory - it's full pathname is `$HOME/.keycloak/kcadm.config` (on Windows it's `%HOMEPATH%\.keycloak\kcadm.config`).
+By default, the Admin CLI automatically maintains a configuration file called [filename]`kcadm.config` located under the user's home directory, which is the full path name [filename]`$HOME/.keycloak/kcadm.config`. On Windows, the full path name is [filename]`%HOMEPATH%\.keycloak\kcadm.config`.
```

Comment:
First off, we are not listing the full path name. There is a variable in there that stands in for /home/username. The true full path name would be /home/username/.keycloak/kcadm.config.

I suggest rephrasing to say this:

By default, the Admin CLI automatically maintains a configuration file called [filename]`kcadm.config` located at [filename]`$HOME/.keycloak/kcadm.config`. On Windows, the file is located at [filename]`%HOMEPATH%\.keycloak\kcadm.config`.

Reply:
It doesn't have to say full path name. 'Located at' sounds good to me.

@tanberry Yes, 'maintained' means it is written to, and updated. As session progresses, the access token reaches it's time to live, and is automatically refreshed. The new one is saved in the file. It can be manually edited because it's a JSON file but I don't see a use case where that would be needed. We don't describe file's format, and we don't instruct users to change it directly. The important thing here is to point out that this file exists, and that it contains some secrets and so it should not be readable to other users.

Reply:
I used the two sentences suggested by @tanberry.